### ISSUE TITLE
Major regression in all E2E tests, Istio fails to startup properly startup

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#17366]
Testgrid of a single E2E showing a major increase: https://k8s-testgrid.appspot.com/istio_istio#e2e-simpleTests_istio&width=20

Seems to correlate to https://github.com/istio/istio/pull/17310 being merged **possibly**.

I am investigating this, will add relevant logs

**Comment by howardjohn** [comment]
Digging into https://prow.istio.io/view/gcs/istio-prow/logs/integ-istioctl-k8s-tests_istio_postsubmit/133

Seems like it fails to start up Istio within 30m. Looking at logs:

Deployment starts at :30

[ingress logs](https://storage.googleapis.com/istio-prow/logs/integ-istioctl-k8s-tests_istio_postsubmit/133/artifacts/kind/istio-testing-control-plane/pods/istio-system_istio-ingressgateway-776f79dc9c-xxh6l_2f57099a-1852-4f26-b4e7-1e4939754580/istio-proxy/0.log) spam the standard `Envoy proxy is NOT ready: config not received from Pilot` for a while, also repeatedly closing the connection to pilot. at :47 (17min later) it finally starts up properly. This points to pilot.

[pilot logs](https://storage.googleapis.com/istio-prow/logs/integ-istioctl-k8s-tests_istio_postsubmit/133/artifacts/kind/istio-testing-control-plane/pods/istio-system_istio-pilot-c8ccf47cc-b9x48_468c9c55-65a9-42f4-b89b-c2088906ae80/discovery/0.log) show the standard `Configuration not synced: first push for ` until :47 then things go normally. This points to galley

[galley logs](https://storage.googleapis.com/istio-prow/logs/integ-istioctl-k8s-tests_istio_postsubmit/133/artifacts/kind/istio-testing-control-plane/pods/istio-system_istio-galley-7bb4d4668b-8zmzr_75f87b8a-7515-4e12-ade2-f6bfe36e8528/galley/0.log) show a bunch of stuff, then a 5 min gap in logs, until at :47 it logs
```
2019-09-24T18:47:05.386485495Z stdout F 2019-09-24T18:47:05.386126Z	error	Unexpected error when reading response body: &net.OpError{Op:"read", Net:"tcp", Source:(*net.TCPAddr)(0xc0005a2000), Addr:(*net.TCPAddr)(0xc0005a2030), Err:(*os.SyscallError)(0xc000b3c0a0)}
2019-09-24T18:47:05.386561289Z stdout F 2019-09-24T18:47:05.386315Z	error	k8s.io/client-go@v10.0.0+incompatible/tools/cache/reflector.go:95: Failed to list *v1beta1.CustomResourceDefinition: Unexpected error &net.OpError{Op:"read", Net:"tcp", Source:(*net.TCPAddr)(0xc0005a2000), Addr:(*net.TCPAddr)(0xc0005a2030), Err:(*os.SyscallError)(0xc000b3c0a0)} when reading response body. Please retry.
2019-09-24T18:47:05.393477287Z stdout F 2019-09-24T18:47:05.393136Z	info	validation	Endpoint istio-system/istio-galley is ready
2019-09-24T18:47:05.395778461Z stdout F 2019-09-24T18:47:05.395575Z	info	validation	istio-galley validatingwebhookconfiguration (re)loaded: 
``

> 	Endpoint istio-system/istio-galley is ready

This suggest galley wasn't ready until :47 which explains why pilot and ingress fail

**Comment by howardjohn** [comment]
@ozevren any idea what would cause this from galley?

What is also strange is it last logs `Checking pods ready...` at :35. This is reasonable since the retry is only for 10 minutes and it started at :25, but I would expect a different failure here -- it seems to hang instead. Also it should retry every 10s, but it only does it once and the stack trace shows its stuck on `kubernetes/typed/core/v1.(*pods).List()` so maybe the kube api-server is completely broken, which is why galley fails as well

**Comment by howardjohn** [comment]
Another example -- happens twice in a single test: https://prow.k8s.io/view/gcs/istio-prow/logs/integ-k8s-114_istio_postsubmit/133 this one doesn't have good logging, but we can see it stalls on pod.List() in both cases -- but sometimes it does get to `Checking pods ready` a few times before this happens.

**Comment by howardjohn** [comment]
Here is a security one that doesn't have the exact same symptoms but could be related: https://prow.k8s.io/view/gcs/istio-prow/logs/integ-security-k8s-tests_istio_postsubmit/133 `secret istio.default did not become existent within 20s`

Another `pod.List()` one: https://prow.k8s.io/view/gcs/istio-prow/logs/integ-galley-k8s-tests_istio_postsubmit/132

Here is an interesting case: https://prow.k8s.io/view/gcs/istio-prow/logs/integ-pilot-k8s-tests_istio_postsubmit/133. We get a slightly different message `panic: test timed out after 30m0s` and this time its *after* we deploy istio

**Comment by howardjohn** [comment]
Here is a new one: [sidecar logs](https://storage.googleapis.com/istio-prow/logs/e2e-bookInfoTests-non-mcp_istio_postsubmit/138/artifacts/demo-test-f752770a90ef4d45b7da81403df/mongodb-v1-59dd775d5d-4ssrr_container:istio-proxy.log) show failed to get LDS config from pilot, but [pilot logs](https://storage.googleapis.com/istio-prow/logs/e2e-bookInfoTests-non-mcp_istio_postsubmit/138/artifacts/demo-test-f752770a90ef4d45b7da81403df/istio-pilot-5f5b84d485-5frsn_container:discovery.log) show they were actually sent (multiple times)!

Whats very confusing is we had a very sudden degradation of all our tests but they all have different symptoms

**Comment by ozevren** [comment]
Looking at [this](https://storage.googleapis.com/istio-prow/logs/integ-istioctl-k8s-tests_istio_postsubmit/133/artifacts/kind/istio-testing-control-plane/pods/istio-system_istio-galley-7bb4d4668b-8zmzr_75f87b8a-7515-4e12-ade2-f6bfe36e8528/galley/0.log), it looks like the call to read CRDs stalled for a very long time. Galley starts around 18:30:57 and the read of CRDs is done around 18:47:06. The watch didn't complete for ~15-17 minutes. This one definitely  looks like an infra/apiserver problem to me.

**Comment by howardjohn** [comment]
@ozevren I agree, seems like some infra problem -- all the failures could be explained by network issues. Thanks for looking at it. 

Seems somehow apiserver calls are getting stalled for indefinite amounts of time, which is weird since I would expect some timeouts at least

**Comment by clarketm** [comment]
@howardjohn
There was a major GCP disruption yesterday (2019-09-24) affecting DNS, GCE, and several other core GCP services. ([ref](https://status.cloud.google.com)). It is likely this *disruption* is the direct cause of the experienced network failures.

**Comment by howardjohn** [comment]
This is resolved -- I have no idea why though.

tl;dr is many calls to api-server timing out. Switching node pools does not help at all. Upgrading GKE version *does* fix the problem. During the same time there were GKE issues with api server timeouts, but our timeouts were to the `kind` api-server. Why the GKE issues could impact that is unanswered, but I think we can close as tests are passing
